pom.xml

```diff
@@ -166,5 +166,71 @@
         <module>drools-examples-android</module>
       </modules>
     </profile>
+    <profile>
+      <id>run-code-coverage</id>
+      <properties>
+        <jacoco.excludes></jacoco.excludes>
+        <jacoco.exec.file>${project.build.directory}/jacoco.exec</jacoco.exec.file>
+        <jacoco.agent.line>-javaagent:${settings.localRepository}/org/jacoco/org.jacoco.agent/${version.jacoco.plugin}/org.jacoco.agent-${version.jacoco.plugin}-runtime.jar=destfile=${jacoco.exec.file},append=true,excludes=${jacoco.excludes}</jacoco.agent.line>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.jacoco</groupId>
+          <artifactId>org.jacoco.agent</artifactId>
+          <classifier>runtime</classifier>
+          <scope>test</scope>
+        </dependency>
+      </dependencies>
+      <build>
+        <pluginManagement>
+          <plugins>
+            <plugin>
```

Comment:
Eventually this entire plugin declaration can be removed if we use JaCoCo in the java agent mode including unit tests. That means you have to pass the agent line property to surefire plugin instead of the reference to the jacoco.exec file and also declare dependency from surefire on the agent. See https://github.com/kiegroup/optaplanner/blob/master/pom.xml#L147